src/config.rs

```diff
@@ -6,13 +6,22 @@ use std::result::Result;
 
 use crate::hash::HashAlgorithm;
 
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct Empty {}
+
 #[derive(Serialize, Deserialize, Debug, PartialEq)]
 pub struct Location {
     // Practically, doing anything with locations (therefore needing
     // access to the "type" and "args" fields) is going to require we
     // know how to deserialise into a union type; for example
     // https://stackoverflow.com/q/66964692
+    //
+    // However, we need to support the 'local' type, which takes no
+    // arguments, so implement enough here to be able to write one.
     pub name: String,
+    #[serde(rename = "type")]
+    pub loc_type: String,
+    pub args: Empty,
```

Comment:
```suggestion
    pub args: HashMap<String, serde_json::Value>,
```

maybe?

The Empty class feels a bit weird, especially if it is ever going to de-serialize a non-empty thing. I guess serde probably ignores unknown fields so it could be fine as is. Up to you